chapters/mitigations-and-defensive-strategies/defense-mechanisms/activities/04-rwslotmachine4/sol/Dockerfile

```diff
@@ -0,0 +1,28 @@
+# Stage 1: Build Stage
+FROM ubuntu:22.04 AS builder
+
+WORKDIR /app
+
+RUN apt update && apt install -y build-essential
+
+# Copy the source code
+COPY rwslotmachine4.c .
+
+# Compile the C program
+RUN gcc -o rwslotmachine4 rwslotmachine4.c
```

Comment:
I would have a `Makefile` that is copied here and then a `make` command would be used.